--- openjdk-jdk-10.0.1+10/src/hotspot/cpu/x86/c1_LIRAssembler_x86.cpp	2018-10-30 00:36:13.093530124 +0000
+++ openjdk-jdk-10.0.1+10/src/hotspot/cpu/x86/c1_LIRAssembler_x86.cpp	2018-10-30 00:36:42.925945642 +0000
@@ -1288,7 +1288,7 @@
     case T_BOOLEAN: // fall through
     case T_BYTE: {
       Register dest_reg = dest->as_register();
-      assert(VM_Version::is_P6() || dest_reg->has_byte_register(), "must use byte registers if not P6");
+//      assert(VM_Version::is_P6() || dest_reg->has_byte_register(), "must use byte registers if not P6");
       if (VM_Version::is_P6() || from_addr.uses(dest_reg)) {
         __ movsbl(dest_reg, from_addr);
       } else {
@@ -1301,7 +1301,7 @@
 
     case T_CHAR: {
       Register dest_reg = dest->as_register();
-      assert(VM_Version::is_P6() || dest_reg->has_byte_register(), "must use byte registers if not P6");
+//      assert(VM_Version::is_P6() || dest_reg->has_byte_register(), "must use byte registers if not P6");
       if (VM_Version::is_P6() || from_addr.uses(dest_reg)) {
         __ movzwl(dest_reg, from_addr);
       } else {

