--- openjdk-jdk-10.0.1+10/make/lib/Lib-jdk.jdwp.agent.gmk.orig	2018-10-19 03:06:39.942275021 +0000
+++ openjdk-jdk-10.0.1+10/make/lib/Lib-jdk.jdwp.agent.gmk	2018-10-19 03:08:01.419559667 +0000
@@ -85,7 +85,7 @@
     LDFLAGS := $(LDFLAGS_JDKLIB) \
         $(call SET_SHARED_LIBRARY_ORIGIN), \
     LIBS := $(JDKLIB_LIBS), \
-    LIBS_linux := $(LIBDL), \
+    LIBS_linux := -liconv $(LIBDL), \
     LIBS_solaris := $(LIBDL), \
     LIBS_macosx := -liconv, \
     LIBS_aix := -liconv, \
