--- openjdk-jdk-9+181/hotspot/src/os/linux/vm/os_linux.cpp.orig	2017-11-30 14:45:15.410288339 +1100
+++ openjdk-jdk-9+181/hotspot/src/os/linux/vm/os_linux.cpp	2017-11-30 16:43:30.944849904 +1100
@@ -36,6 +36,7 @@
 #include "memory/allocation.inline.hpp"
 #include "memory/filemap.hpp"
 #include "oops/oop.inline.hpp"
+#include "os_linux.hpp"
 #include "os_linux.inline.hpp"
 #include "os_share_linux.hpp"
 #include "prims/jniFastGetField.hpp"
@@ -4805,7 +4806,7 @@
   }
   Linux::capture_initial_stack(JavaThread::stack_size_at_create());
 
-#if defined(IA32)
+#if defined(IA32) && !defined(__UCLIBC__)
   workaround_expand_exec_shield_cs_limit();
 #endif
 
