--- openjdk-12.0.1+12/make/lib/Awt2dLibraries.gmk	2018-10-18 00:41:23.011330178 +0000
+++ openjdk-12.0.1+12/make/lib/Awt2dLibraries.gmk	2018-10-18 00:42:30.387994505 +0000
@@ -922,7 +922,7 @@
     LIBSPLASHSCREEN_LDFLAGS := -delayload:user32.dll
     LIBSPLASHSCREEN_LIBS += kernel32.lib user32.lib gdi32.lib delayimp.lib $(WIN_JAVA_LIB) jvm.lib
   else
-    LIBSPLASHSCREEN_LIBS += $(X_LIBS) -lX11 -lXext $(LIBM) -lpthread -ldl
+    LIBSPLASHSCREEN_LIBS += $(X_LIBS) -lX11 -lXext $(LIBM) -lpthread -ldl -liconv
   endif
 
   $(eval $(call SetupNativeCompilation,BUILD_LIBSPLASHSCREEN, \
--- openjdk-12.0.1+12/make/lib/Lib-java.instrument.gmk	2018-10-18 00:43:20.764509642 +0000
+++ openjdk-12.0.1+12/make/lib/Lib-java.instrument.gmk	2018-10-18 00:43:56.272881312 +0000
@@ -69,7 +69,7 @@
     LDFLAGS_aix := -L$(SUPPORT_OUTPUTDIR)/native/java.base, \
     LIBS := $(JDKLIB_LIBS), \
     LIBS_unix := -ljava -ljvm $(LIBZ), \
-    LIBS_linux := -ljli $(LIBDL), \
+    LIBS_linux := -liconv -ljli $(LIBDL), \
     LIBS_solaris := -ljli $(LIBDL), \
     LIBS_aix := -liconv -ljli_static $(LIBDL), \
     LIBS_macosx := -liconv -framework Cocoa -framework Security \
