--- openjdk-12.0.1+12/src/java.desktop/share/native/libjavajpeg/jdphuff.c	2018-05-15 07:04:10.272822750 +0000
+++ openjdk-12.0.1+12/src/java.desktop/share/native/libjavajpeg/jdphuff.c	2018-05-15 23:35:04.923841195 +0000
@@ -202,9 +202,11 @@
  * On some machines, a shift and add will be faster than a table lookup.
  */
 
+#define AVOID_TABLES
 #ifdef AVOID_TABLES
 
-#define HUFF_EXTEND(x,s)  ((x) < (1<<((s)-1)) ? (x) + (((-1)<<(s)) + 1) : (x))
+#define NEG_1 ((unsigned)-1)
+#define HUFF_EXTEND(x,s)  ((x) < (1<<((s)-1)) ? (x) + (((NEG_1)<<(s)) + 1) : ((unsigned)x))
 
 #else
 
@@ -502,7 +504,7 @@
   phuff_entropy_ptr entropy = (phuff_entropy_ptr) cinfo->entropy;
   int Se = cinfo->Se;
   int p1 = 1 << cinfo->Al;      /* 1 in the bit position being coded */
-  int m1 = (-1) << cinfo->Al;   /* -1 in the bit position being coded */
+  int m1 = (NEG_1) << cinfo->Al;   /* -1 in the bit position being coded */
   register int s, k, r;
   unsigned int EOBRUN;
   JBLOCKROW block;
