--- openjdk-12.0.1+12/src/hotspot/share/runtime/java.cpp	2018-10-22 03:49:03.035613787 +0000
+++ openjdk-12.0.1+12/src/hotspot/share/runtime/java.cpp	2018-10-22 03:05:43.001767411 +0000
@@ -672,6 +672,11 @@
   assert(!_current.is_valid(), "Don't initialize twice");
 
   void *lib_handle = os::native_java_library();
+  assert(lib_handle != NULL, "Cannot find Native Java Library.");
+
+  void *dll_lookup = os::dll_lookup(lib_handle, "JDK_GetVersionInfo0");
+  assert(dll_lookup != NULL, "DLL_Lookup failed to find JDK_GetVersionInfo0");
+
   jdk_version_info_fn_t func = CAST_TO_FN_PTR(jdk_version_info_fn_t,
      os::dll_lookup(lib_handle, "JDK_GetVersionInfo0"));
 
